sway-lsp/src/lib.rs

```diff
@@ -1,23 +1,81 @@
 #![recursion_limit = "256"]
 
-use tower_lsp::{LspService, Server};
-
 mod capabilities;
 pub mod config;
 mod core;
 pub mod error;
+pub mod global_state;
+mod handlers {
+    pub mod notification;
+    pub mod request;
+}
+pub mod lsp_ext;
 pub mod server;
 mod traverse;
 pub mod utils;
-use server::Backend;
 
-pub async fn start() {
-    let stdin = tokio::io::stdin();
-    let stdout = tokio::io::stdout();
+use global_state::GlobalState;
+use lsp_types::{
+    CodeActionProviderCapability, CodeLensOptions, CompletionOptions, ExecuteCommandOptions,
+    HoverProviderCapability, OneOf, RenameOptions, SemanticTokensFullOptions, SemanticTokensLegend,
+    SemanticTokensOptions, ServerCapabilities, TextDocumentSyncCapability, TextDocumentSyncKind,
+    WorkDoneProgressOptions,
+};
+use tower_lsp::{LspService, Server};
 
-    let (service, socket) = LspService::build(Backend::new)
-        .custom_method("sway/show_ast", Backend::show_ast)
-        .custom_method("textDocument/inlayHint", Backend::inlay_hints)
+pub async fn start() {
+    let (service, socket) = LspService::build(GlobalState::new)
+        .custom_method("sway/show_ast", GlobalState::show_ast)
```

Comment:
It's a little funky to have the custom capability handlers be part of `GlobalState`.  Might be better to put them in their own module under the handlers directory, like `custom.rs` or `lsp_ext.rs`. Or even put them in with `request.rs` since they are request handlers.

Reply:
Yeah I agree, this is just a temporary work around as tower-lsp requires that these methods are added to the type `LanguageServer` has been implemented for. In this case it's `GlobalState` this won't be the case once we move to the new event loop.